plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -881,6 +883,111 @@ public void testSystemSchemas()
         assertQueryReturnsEmptyResult("SHOW SCHEMAS IN mongodb LIKE 'local'");
     }
 
+    @Test(dataProvider = "nestedValuesProvider")
+    public void testFiltersOnDereferenceColumnReadsLessData(String expectedValue, String expectedType)
```

Comment:
Why not define this test in `BaseMongoConnectorSmokeTest`?

Reply:
I learned from @findepi that `ConnectorTest` should have `>=` coverage than `ConnectorSmokeTest`. I have added `testProjectionPushdownMultipleRows` in `BaseMongoConnectorSmokeTest` to cover smoke test.

I will add a `filters on dereference column` test in  `BaseMongoConnectorSmokeTest` as well.

Reply:
I learned from @findepi that `ConnectorTest` should have `>=` coverage than `ConnectorSmokeTest`. I have added `testProjectionPushdownMultipleRows` in `BaseMongoConnectorSmokeTest` to cover smoke test.

I will add a `filters on dereference column` test in  `BaseMongoConnectorSmokeTest` as well.